test/test_view/test_fileproxy.py

```diff
@@ -208,3 +208,146 @@ def test_proxy_key_not_found(self, create_connection):
+        owner = project.owner
+
+        signature = signer.dumps({'task_id': 100})
+
+        url = '/fileproxy/hdfs/test/%s/file.pdf?api_key=%s&task-signature=%s' \
+            % (project.id, owner.api_key, signature)
+        with patch.dict(self.flask_app.config, self.app_config):
+            res = self.app.get(url, follow_redirects=True)
+            assert res.status_code == 400, res.status_code
+
+    @with_context
+    @patch('pybossa.view.fileproxy.HDFSKerberos.get')
+    @patch('pybossa.view.fileproxy.requests.get')
+    def test_proxy_owner(self, http_get, hdfs_get):
+        res = MagicMock()
+        res.json.return_value = {'key': 'testkey'}
+        http_get.return_value = res
+
+        project = ProjectFactory.create(info={
+            'ext_config': {
+                'encryption': {'key_id': 123}
+            }
+        })
+        url = '/fileproxy/hdfs/test/%s/file.pdf' % project.id
+        task = TaskFactory.create(project=project, info={
+            'url': url
+        })
+        owner = project.owner
+
+        signature = signer.dumps({'task_id': task.id})
+        req_url = '%s?api_key=%s&task-signature=%s' % (url, owner.api_key, signature)
+
+        encryption_key = 'testkey'
+        aes = AESWithGCM(encryption_key)
+        hdfs_get.return_value = aes.encrypt('the content')
+
+        with patch.dict(self.flask_app.config, self.app_config):
+            res = self.app.get(req_url, follow_redirects=True)
+            assert res.status_code == 200, res.status_code
+            assert res.data == 'the content', res.data
+
+    @with_context
+    @patch('pybossa.view.fileproxy.HDFSKerberos.get')
+    @patch('pybossa.view.fileproxy.requests.get')
+    def test_proxy_admin(self, http_get, hdfs_get):
+        res = MagicMock()
+        res.json.return_value = {'key': 'testkey'}
+        http_get.return_value = res
+
+        admin, owner = UserFactory.create_batch(2)
+        project = ProjectFactory.create(owner=owner, info={
+            'ext_config': {
+                'encryption': {'key_id': 123}
+            }
+        })
+        url = '/fileproxy/hdfs/test/%s/file.pdf' % project.id
+        task = TaskFactory.create(project=project, info={
+            'url': url
+        })
+
+        signature = signer.dumps({'task_id': task.id})
+        req_url = '%s?api_key=%s&task-signature=%s' % (url, admin.api_key, signature)
+
+        encryption_key = 'testkey'
+        aes = AESWithGCM(encryption_key)
+        hdfs_get.return_value = aes.encrypt('the content')
+
+        with patch.dict(self.flask_app.config, self.app_config):
+            res = self.app.get(req_url, follow_redirects=True)
+            assert res.status_code == 200, res.status_code
+            assert res.data == 'the content', res.data
+
+    @with_context
+    @patch('pybossa.view.fileproxy.HDFSKerberos.get')
+    @patch('pybossa.view.fileproxy.requests.get')
+    def test_proxy_key_err(self, http_get, hdfs_get):
```

Comment:
hdfs_get unused here?

Reply:
it is used to avoid making the actual request